presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java

```diff
@@ -14,21 +14,30 @@
 package io.prestosql.spi.security;
 
 import java.security.Principal;
+import java.util.Map;
 import java.util.Optional;
 
+import static java.util.Collections.emptyMap;
 import static java.util.Objects.requireNonNull;
 
 public class ConnectorIdentity
 {
     private final String user;
     private final Optional<Principal> principal;
     private final Optional<SelectedRole> role;
+    private final Map<String, String> credentials;
 
     public ConnectorIdentity(String user, Optional<Principal> principal, Optional<SelectedRole> role)
+    {
+        this(user, principal, role, emptyMap());
+    }
+
+    public ConnectorIdentity(String user, Optional<Principal> principal, Optional<SelectedRole> role, Map<String, String> credentials)
     {
         this.user = requireNonNull(user, "user is null");
         this.principal = requireNonNull(principal, "principal is null");
         this.role = requireNonNull(role, "role is null");
+        this.credentials = requireNonNull(credentials, "credentials is null");
```

Comment:
ImmutableMap.copyOf

Reply:
This is in the SPI, so we can't use Guava. However, we should make a defensive copy:
```java
this.credentials = unmodifiableMap(new HashMap<>(requireNonNull(credentials, "credentials is null")));